packages/core/src/io/node-io.ts

```diff
@@ -34,31 +34,49 @@ import { PlatformIO } from './platform-io';
 export class NodeIO extends PlatformIO {
 	private _fs;
 	private _path;
+	private _fetch;
+	private _httpRegex = /https?:\/\//;
+	public useFetch = false;
 
 	/** Constructs a new NodeIO service. Instances are reusable. */
-	constructor() {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	constructor(fetch?: any) {
 		super();
 		// Excluded from browser builds with 'package.browser' field.
 		this._fs = require('fs').promises;
 		this._path = require('path');
+		this._fetch = fetch;
+    this.useFetch = !!fetch;
 	}
 
 	protected async readURI(uri: string, type: 'view'): Promise<Uint8Array>;
 	protected async readURI(uri: string, type: 'text'): Promise<string>;
 	protected async readURI(uri: string, type: 'view' | 'text'): Promise<Uint8Array | string> {
 		switch (type) {
-			case 'view':
+			case 'view': {
+        if(this.useFetch && this._httpRegex.exec(uri)) {
```

Comment:
minor: Let's check httpRegex first, and if it matches but `useFetch` is disabled we can throw a clear error. Avoids trying to read a bad path on disk like #452 and tells the user how to fix it.